src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClones = [Object.assign({}, state)];
+
+  for (const action of actions) {
+    let newState = Object.assign({}, stateClones[stateClones.length - 1]);
+
+    switch (action.type) {
+      case 'addProperties':
+        if (typeof action.extraData === 'object') {
```

Comment:
you don't need make this check since if type is 'addProperties' the sectonde argument will be always an object